components/ui/TextInput.tsx

```diff
@@ -0,0 +1,27 @@
+import { FC } from "react"
+
+interface Props {
+    inputName: string,
+    className?: string,
+}
+
+export const TextInput:FC<Props>= ({ inputName, className }) => {
+    return (
+        <div className={"flex flex-col flex-grow"}>
+            <label
+                htmlFor={inputName}
+                className="block mb-2 text-sm text-gray-800 dark:text-white"
+            >
+                {inputName.charAt(0).toUpperCase() + inputName.slice(1)}
+            </label>
+            <input
+                type="text"
+                id={inputName}
+                className={"bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg " + 
```

Comment:
hah! you can use string templates to avoid having to use concatenation to break this into multiple lines